tuf/ngclient/fetcher.py

```diff
@@ -20,7 +29,7 @@ class FetcherInterface:
     __metaclass__ = abc.ABCMeta
 
     @abc.abstractmethod
-    def fetch(self, url, required_length):
+    def fetch(self, url: str, required_length: int) -> Iterator[bytes]:
```

Comment:
maybe should be max_length (also docstring could be better)?

This is sort of API already but the change maybe worth it as required_length really is misleading...